File: imagepicker/src/main/java/com/esafirm/imagepicker/adapter/BaseListAdapter.java
package com.esafirm.imagepicker.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;

import com.esafirm.imagepicker.features.imageloader.ImageLoader;

public abstract class BaseListAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T> {

    private final Context context;
    private final LayoutInflater inflater;
    private final ImageLoader imageLoader;

    public BaseListAdapter(Context context, ImageLoader imageLoader) {
        this.context = context;
        this.inflater = LayoutInflater.from(context);
        this.imageLoader = imageLoader;
    }

    public ImageLoader getImageLoader() {
        return imageLoader;
    }

    public Context getContext() {
        return context;
    }

    public LayoutInflater getInflater() {
        return inflater;
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/adapter/BaseListAdapter.kt
package com.esafirm.imagepicker.adapter

import android.content.Context
import android.view.LayoutInflater
import androidx.recyclerview.widget.RecyclerView
import com.esafirm.imagepicker.features.imageloader.ImageLoader

abstract class BaseListAdapter<T : RecyclerView.ViewHolder>(
    val context: Context,
    val imageLoader: ImageLoader
) : RecyclerView.Adapter<T>() {
    val inflater: LayoutInflater = LayoutInflater.from(context)
}